src/api/users.ts

```diff
@@ -0,0 +1,24 @@
+import firebase from "firebase/app";
+
+import { UserWithLocation } from "types/User";
+
+import { WithId } from "utils/id";
+
+export const fetchUsersRecord = async (): Promise<
+  Record<string, WithId<UserWithLocation>>
+> =>
+  firebase
+    .firestore()
+    .collection("users")
+    .get()
```

Comment:
This will still fetch literally all users every time, which is hugely inefficient. We want to query for only new changes since we last updated.